scipy/signal/tests/test_peak_finding.py

```diff
@@ -448,13 +458,28 @@ def test_raises(self):
         with raises(ValueError, match='index'):
             # empty x with peaks supplied
             peak_widths([], [1, 2])
-        with raises(ValueError, match='integers'):
-            # peak is not of subtype int
+        with raises(TypeError, match='Cannot safely cast'):
+            # peak cannot be safely casted to intp
             peak_widths(np.arange(10), [1.1, 2.3])
         with raises(ValueError, match='rel_height'):
             # rel_height is < 0
             peak_widths(np.arange(10), [1, 2], rel_height=-1)
 
+        # With mismatching peak & prominence data
+        x = [0, 1, 0]
+        peak = [1]
+        cases = [
+            # (prominence, left base, right base)
+            (1., -1, 2),  # left base not in x
+            (1., 0, 3),  # right base not in x
+            (1., 1, 2),  # left base same as peak
+            (1., 0, 1)  # right base same as peak
+        ]
+        for case in cases:
```

Comment:
Iterating over the cases might signal that a parametrized pytest decorator might be suitable?

Reply:
I placed it with the other exceptions checks because it is essentially one. But if a parametrized pytest is the preferred way I'll move it to its own function.